### ISSUE TITLE
CA1000 should not fire for conversion operators on static classes

### ISSUE THREAD

**Issue by dpaoliello** [dotnet/roslyn-analyzers#1287]
#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

DoNotDeclareStaticMembersOnGenericTypes

#### Repro steps

```csharp
public class Class1<T>
{
    public static implicit operator Class1<T>(T value) => new Class1<T>();
}
```

#### Expected behavior

CA1000 is not raised - there is no way to write this conversion that would permit type inference.

#### Actual behavior

CA1000 is raised

**Comment by mavasani** [comment]
@ivanbasov Can you take this up? Looks similar to your fix https://github.com/dotnet/roslyn-analyzers/pull/1329 but for CA1000
